manifests/params.pp

```diff
@@ -30,6 +30,12 @@
   ]
   $purge_plugins = false
 
+  if versioncmp(pick($facts['jenkins_version'], '0'), '2.313') >= 0 {
+    $systemd_type = 'simple'
+  } else {
+    $systemd_type = 'forking'
+  }
```

Comment:
Wouldn't it be safer to assume `simple` and only if the fact is an older version to return `forking`?

During the initial run the fact won't be available (since the war is not present) so you'll always need 2 runs.

I would also be happy to skip automatic detection and require the admin to select `forking` if they're on an older version. Even the current LTS should default to simple and they don't support older versions. This saves a lot of computing power (starting up Java just to get a version number) that most people won't need.

Reply:
Yes during the initial setup that not work for the last release.

I can change like that if you want:
`if versioncmp(pick($facts['jenkins_version'], '2.313'), '2.313') >= 0 {`

That just create a breaking change if the libdir is changed and the fact not resolve

Reply:
I just wrote a dynamic solution but we can remove the fact if you want

```
time /bin/java -jar /usr/lib/jenkins/jenkins.war --version
2.319.1

real	0m0.371s
user	0m0.562s
sys	0m0.040s
```

Reply:
We can also remove the `--deamon` and set simple options for all jenkins versions

Reply:
I was actually thinking about swapping the logic so it's:
```suggestion
  if versioncmp(pick($facts['jenkins_version'], '0'), '2.313') < 0 {
    $systemd_type = 'forking'
  } else {
    $systemd_type = 'simple'
  }
```

That way it defaults to `simple` if the fact is not found.

However, in general I'm happier with looking to the future and aligning on `simple`, removing `forking` altogether. With that I will note that I haven't looked at this area so I may be missing something.

Reply:
I haven't really watched either... In fact I don't understand why the package doesn't create the unit file and why in Debian family they use a init script... And I don't know if it's work with Debian family

For imformation `--deamon` option is set in the `/etc/sysconfig/jenkins` but it's useless since `2.313`

I can try on a Ubuntu 20.04 this week or next week If you want

Reply:
It's actually an excellent question and I wondered the same thing. Perhaps we should actually file an upstream bug. It may just be an oversight.

Reply:
https://github.com/jenkinsci/packaging/issues/216